tests/scenarios.go

```diff
@@ -406,13 +409,29 @@ func getRequestBuilder(ctx gobdd.Context) (reflect.Value, error) {
 
 	// first argument is always context.Context
 	in[0] = reflect.ValueOf(GetCtx(ctx))
+	requestArgs := GetRequestArguments(ctx)
 	for i := 1; i < f.Type().NumIn(); i++ {
-		object := GetRequestArguments(ctx)[i-1]
-		in[i] = object.(reflect.Value)
+		if len(requestArgs) > i-1 {
+			object := requestArgs[i-1]
+			in[i] = object.(reflect.Value)
+		} else {
+			return reflect.Value{}, fmt.Errorf("Error getting args, this is likely a path argument.")
```

Comment:
I've thought about it a bit more. Instead of returning an error that forces us to check the error message to make a decision later on, which doesn't sit right with me, can we directly return `f` here, with a `nil` error, and later check the `Kind` of the returned `Value` ? 
If we've returned a value of kind `Func` (we haven't called it yet as we still need some arguments), then it means we still need to process args, otherwise I think the result is of Kind `Struct` since it's the actual "builder" object (result of calling `f`)